src/Converter/ShipmentLineItemsConverter.php

```diff
@@ -50,14 +57,28 @@ private function convertUnitRefundToLineItem(ShipmentRefund $shipmentRefund): Li
         Assert::isInstanceOf($shipment, AdjustableInterface::class);
         Assert::lessThanEq($shipmentRefund->total(), $shipment->getAdjustmentsTotal());
 
+        $grossValue = $shipmentRefund->total();
+
+        $taxAdjustments = $shipment->getAdjustments(AdjustmentInterface::TAX_ADJUSTMENT);
+
+        $taxAdjustmentAmount = 0;
+        if (count($taxAdjustments) > 0) {
+            $taxAdjustmentAmount = $this->taxRateAmountProvider->provide($taxAdjustments->first());
+        }
+
+        $taxRate = $taxAdjustmentAmount * 100 . '%';
+        $taxAmount = (int) ($grossValue * $taxAdjustmentAmount);
```

Comment:
I guess you can use the value from tax adjustments instead of calculating the tax amount again